docs/Writing_yeb_easyconfig_files.rst

```diff
@@ -0,0 +1,206 @@
+It's worth noting that there's a subtle difference between using single and double quotes, see
+`Flow Scalar Styles <http://www.yaml.org/spec/1.2/spec.html#id2786942>`_.
+
+Examples::
+
+    name: gzip
+    version: 1.6
+
+    # single quotes are required for string values representing URLs, to escape the ':'
+    homepage: 'http://www.gnu.org/software/gzip/'
+
+    parallel: 1
+
+Multiline strings can be expressed using indentation::
+
+    description:
+        gzip is a popular data compression program
+        as a replacement for compress
+
+Sequences
+#########
+
+Sequence values (a.k.a. lists) can be expressed in different ways, depending on their size.
+
+If there are a limited number of (short) entries the value can be expressed on a single line,
+using square brackets '``[``' '``]``' and with comma '``,``' as separator.
+
+Example::
+
+    # quotes are required to escape the ':'
+    source_urls: ['http://ftpmirror.gnu.org/gzip/', 'ftp://ftp.gnu.org/gnu/gzip/']
+
+Alternatively indentation can be used for scope, with each entry on its own line,
+indicated with a dash and a space ``- ``.
+
+Example::
+
+    # no quotes required here, since there's no ambiguity w.r.t. ':'
+    source_urls:
+        - http://ftpmirror.gnu.org/gzip/
+        - http://ftp.gnu.org/gnu/gzip/
+        - ftp://ftp.gnu.org/gnu/gzip/
+
+Mappings
+########
+
+Mapping values (a.k.a. dictionaries) are expressed using a colon '``:``' and space as key-value separator,
+a comma '``,``' to key-value pairs, and curly braces '``{``' '``}``' to mark the start/end.
+
+For example::
+
+    toolchain: {name: intel, version: 2015b}
+
+Nesting
+#######
+
+Different types of values can be nested.
+
+For example, sequence values can be used in a mapping::
+
+    sanity_check_paths: {
+        files: [bin/gunzip, bin/gzip, bin/uncompress],
+        dirs: [],
+    }
+    
+
+.. _easyconfig_yeb_format_syntax_template_values_constants:
+
+Templates values and constants
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Template values can be specified as a part of string values, using ``%(template_name)``.
+
+Template constants are injected by the easyconfig ``.yeb`` parser as *node anchors*,
+and can be referred to with an *alias node*, i.e. using an asterisk ``*``.
+
+For example::
+
+    source_urls: [*GNU_SOURCE]
+    sources: ['%(name)s-%(version)s.tar.gz']  # equivalent with [*SOURCE_TAR_GZ]
+
+See also :ref:`easyconfig_param_templates`.
+
+.. _easyconfig_yeb_format_syntax_dependencies:
+
+Dependencies
+~~~~~~~~~~~~
+
+*(WORK IN PROGRESS)*
+
+Example::
+
+    dependencies:
+        - 
+
+.. _easyconfig_yeb_format_examples:
+
+Examples
+--------
+
+gzip v1.6 with ``GCC/4.9.2`` toolchain
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+    %YAML 1.2
+    ---
+    easyblock: ConfigureMake
+
+    name: gzip
+    version: '1.6'  # FIXME bug: quotes are required here to make sure this is parsed a a string, not a floating point value
```

Comment:
@Caylo: please look into this, make very sure somehow that `version` is string after parsing...

It's probably time we define the expected type for each of the easyconfig parameters...

Also, please include this `gzip.yeb` in the test easyconfigs, and include it in the tests for `.yeb`.